Starforge/Mod/EntityRegistry.cs

```diff
@@ -0,0 +1,92 @@
+﻿using Starforge.Core;
+using Starforge.Map;
+using Starforge.Mod.API;
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+
+namespace Starforge.Mod {
+    public static class EntityRegistry {
+        private static List<Placement> EntityPlacements;
+        private static Dictionary<string, EntityCreator> EntityCreators;
+        private static Dictionary<Type, PropertyInfo[]> EntityCache;
+
+        static EntityRegistry() {
+            EntityPlacements = new List<Placement>();
+            EntityCreators = new Dictionary<string, EntityCreator>();
+            EntityCache = new Dictionary<Type, PropertyInfo[]>();
+        }
+
+        public static Entity CreateEntity(MapElement el, Room room) {
+            EntityData data = new EntityData(el);
+
+            if (EntityCreators.ContainsKey(el.Name)) {
+                return EntityCreators[el.Name](data, room);
+            } else {
+                return new UnknownEntity(data, room);
+            }
+        }
+
+        public static void Register(Type type) {
+            try {
+                if (!type.IsSubclassOf(typeof(Entity))) return;
+
+                EntityDefinitionAttribute attr = type.GetCustomAttribute<EntityDefinitionAttribute>();
+                if (attr != null) {
```

Comment:
Inverting this and the next two checks of this kind and returning immediately can save a lot of indentation